Src/IronPython/Compiler/Ast/Comprehension.cs

```diff
@@ -237,6 +237,27 @@ internal class ComprehensionScope : ScopeStatement {
             return MSAst.Expression.Call(null, typeof(PythonOps).GetMethod(nameof(PythonOps.GetClosureTupleFromContext)), _comprehension.Parent.LocalContext);
         }
 
+        internal override bool TryBindOuter(ScopeStatement from, PythonReference reference, out PythonVariable variable) {
+            ContainsNestedFreeVariables = true;
+            if (TryGetVariable(reference.Name, out variable) && variable.Kind != VariableKind.Nonlocal) {
```

Comment:
There should be no nonlocals in a comprehension, so maybe use an explicit assertion? (see `BindReference` below).